src/jit/flowgraph.cpp

```diff
@@ -22858,10 +22858,9 @@ void Compiler::fgInsertInlineeBlocks(InlineInfo* pInlineInfo)
     BasicBlock*  iciBlock = pInlineInfo->iciBlock;
     BasicBlock*  block;
 
-    // We can write better assert here. For example, we can check that
-    // iciBlock contains iciStmt, which in turn contains iciCall.
     noway_assert(iciBlock->bbTreeList != nullptr);
     noway_assert(iciStmt->gtStmtExpr != nullptr);
+    noway_assert(iciBlock->Contains(iciStmt) && (iciStmt->gtStmtExpr == iciCall));
```

Comment:
I agree with @sandreenko on this - this should be simply an `assert`, since it would be somewhat expensive to always check this in a release build (note that `noway_assert`s are enabled even in release builds, while `assert`s are debug/checked only).